.eslintrc.json

```diff
@@ -0,0 +1,109 @@
+{
+  "env": {
+    "browser": true,
+    "node": true,
+    "jest": true
+  },
+
+  "plugins": ["import"],
+
+  "settings": {
+    "import/parsers": {
+      "@typescript-eslint/parser": [".ts", ".tsx"]
+    },
+    "import/resolver": {
+      "typescript": {
+        "alwaysTryTypes": true
+      },
+      "node": {
+        "extensions": [".js", ".jsx", ".ts", ".tsx", ".json"]
+      }
+    }
+  },
+
+  "overrides": [
+    {
+      "plugins": ["@typescript-eslint"],
+      "parser": "@typescript-eslint/parser",
+      "files": ["**/*.ts"],
+      "excludedFiles": "*.js",
```

Comment:
This isn't consisent with the tsconfig as we are allowing js. We could go either way with this:
- allow js in tsconfig and include js files in linting
- not allow js files in tsconfig and exclude js files in linting

Reply:
this config works for typescript and javascript, javascript is handled by the default then we provide overrides that only work on typescript.

Guess probably we would want to disable javascript entirely though if possible?

Reply:
Ah originally I was thinking that js files wouldn't be linted at all, but this project also uses prettier, so I guess it's partially picked up by that!

I am pro disabling javascript 🔥